apps/website/src/components/logo.tsx

```diff
@@ -0,0 +1,56 @@
+import Image from 'next/image'
+import { useRouter } from 'next/router'
+import { match, P } from 'ts-pattern'
+
+import logoSrc from '../../public/images/logo/default.svg'
+import logoDevSrc from '../../public/images/logo/dev.svg'
+import logoLearnSrc from '../../public/images/logo/learn.svg'
+
+type Props = {
+  label?: boolean
+}
+
+export const Logo = (props: Props) => {
+  const { label = true } = props
+
+  const { pathname } = useRouter()
+
+  return (
+    <div className="flex flex-shrink-0 items-center gap-2">
+      {match(pathname)
+        .with(
+          P.when(p => p.startsWith('/insights')),
+          () => <Image src={logoDevSrc} alt="Status logo" />
+        )
+        .with(
+          P.when(p => p.startsWith('/learn')),
+          () => <Image src={logoLearnSrc} alt="Status logo" />
+        )
+        .with(
+          P.when(p => p.startsWith('/blog')),
+          () => <Image src={logoSrc} alt="Status logo" />
+        )
+        .otherwise(() => (
+          <Image src={logoSrc} alt="Status logo" />
+        ))}
```

Comment:
Never used this approach before. Cool.

Reply:
Yeah, I wanted to try this out. It's a TypeScript implementation of the [JavaScript pattern-matching proposal](https://github.com/tc39/proposal-pattern-matching). I loved it in Rust and eagerly waiting for this to land in JS.

Reply:
Cool shit!